data/scenarios/Challenges/Ranching/gated-paddock/meandering-sheep.sw

```diff
@@ -0,0 +1,36 @@
+// A "sheep" that wanders around randomly.
+
+let forever : cmd unit -> cmd unit = \c. c ; forever c in
+let repeat : int -> cmd unit -> cmd unit =
+  \n. \c. if (n == 0) {} {c ; repeat (n-1) c} in
+let randdir : cmd dir =
+  d <- random 4;
+  return (
+    if (d == 0) {north}
+    {if (d == 1) {east}
+    {if (d == 2) {south} {west}}})
+  in
+
+forever (
+  n <- random 30;
+  wait (30 + n);
+  d <- randdir;
+  turn d;
+  dist <- random 3;
+  repeat dist (
+    b <- blocked;
+    if b {} {
+      move;
```

Comment:
The sheep could crash in rare cases, consider atomic:
```suggestion
  atomic (b <- blocked; if b {} {move});
```

Reply:
I tried adding `atomic`, but now the blocked-checking logic is larger, and the compiler refuses to allow atomic around it.

Reply:
@kostmo AFAIK, we do not allow definitions. If you inline it then it should be fine, because only the move is observable.

Reply:
@kostmo did you try inlining the functions? I don't care as much about the sheep crashing into a fence as I would like to know if the `atomic` command is usable. :wink: